### PR TITLE
[Close #7480] Add new Style::NestedInterpolation cop

### LINKED ISSUE #7480: Not to use nested string interpolation
Hello, 

What do you think about adding a cop which can find examples of using nested interpolation (when we use string interpolation into another string interpolation). Just for me, this is a good example of the bad design and places which should be refactored.

Example,

```ruby
def say_hello(user = nil)
  puts "Hello #{ user.present? ? "dear #{user.first_name}" : "everyone" }. How are you?"
end
```

Thank you!

### PR DESCRIPTION
**Pull request by roman-dubrovsky**
Fix #7480 

This PR adds new `Style::NestedInterpolation` cop which checks so that developers don't use interpolation block into another in interpolation blocks. Also, this PR fixes some issues which related to new cop -  we had nested interpolation in the next two places:

- https://github.com/rubocop-hq/rubocop/blob/2ec092a013e11be5a6f7475623d70a7a44e6abcc/lib/rubocop/cop/style/parallel_assignment.rb#L253
- https://github.com/rubocop-hq/rubocop/blob/2ec092a013e11be5a6f7475623d70a7a44e6abcc/lib/rubocop/cop/style/parallel_assignment.rb#L268

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -3176,6 +3176,11 @@ Style/NegatedWhile:
   Enabled: true
   VersionAdded: '0.20'
 
+Style/NestedInterpolation:
+  Description: 'Avoid using nested interpolation.'
+  Enabled: true
```
Comment (bbatsov) [inline config/default.yml]: New cops should be "pending" by default.

File: lib/rubocop/cop/style/nested_interpolation.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks nested interpolations
```
Comment (bbatsov) [inline lib/rubocop/cop/style/nested_interpolation.rb]: I think you can expand the documentation of this cop with more rationale and examples that are following the same conventions as those in existing cops. Might be a good idea to link the cop to a style guide entry.

File: lib/rubocop/cop/style/nested_interpolation.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      #
+      # @example
+      #
+      #   # bad
+      #   "Hello, #{user.blank? ? 'guest' : "dear #{user.name}"}"
+      #
+      #   # good
+      #   user_name = user.blank? ? 'guest' : "dear #{user.name}"
+      #   "Hello, #{user_name}"
+      class NestedInterpolation < Cop
```
Comment (bbatsov) [inline lib/rubocop/cop/style/nested_interpolation.rb]: I guess this should named `RedundantNestedInterpolation` to be aligned with the typical naming scheme we employ.

File: lib/rubocop/cop/style/nested_interpolation.rb (hunk 2 above)
Comment (roman-dubrovsky) [reply lib/rubocop/cop/style/nested_interpolation.rb]: @bbatsov I've updated the description a little bit but I'm not sure that this is enough. (But the style should be the same as in existing cops). Also, added one more example.

> Might be a good idea to link the cop to a style guide entry.

Does it mean that we need to add it to the style guide? And can you guide me on how to do it?

File: lib/rubocop/cop/style/redundant_nested_interpolation.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for nested interpolation expressions
+      # which can be splitted.
```
Comment (bbatsov) [inline lib/rubocop/cop/style/redundant_nested_interpolation.rb]: can be rewritten to avoid it.

**Comment by bbatsov** [comment]
On second thought - this doesn't actually seem like redundant nested interpolation, so probably `NestedInterpolation` is a better name for the cop. I mean you need it for the code to work the way you want it to. Probably the message should also say `Avoid the use of nested interpolation.` or something along those lines. 

> Does it mean that we need to add it to the style guide? And can you guide me on how to do it?

Just file a PR here https://github.com/rubocop-hq/ruby-style-guide

**Comment by bbatsov** [comment]
I'm closing this PR due to no recent activity. Feel free to re-open it if you ever come back to it.

**Pull request closed without merging**